src/main/java/oryanmoshe/kafka/connect/util/TimestampConverter.java

```diff
@@ -85,53 +107,102 @@ public void converterFor(RelationalColumn column, ConverterRegistration<SchemaBu
                     else if (column.hasDefaultValue()) {
                         return column.defaultValue();
                     }
-                    return rawValue;
+                    return null;
                 }
 
-                Long millis = getMillis(rawValue.toString(), isTime);
-                if (millis == null)
-                    return rawValue.toString();
+                String stringValue = rawValue.toString();
 
-                Instant instant = Instant.ofEpochMilli(millis);
-                Date dateObject = Date.from(instant);
                 if (this.debug)
                     System.out.printf(
-                            "[TimestampConverter.converterFor] Before returning conversion. column.name: %s, column.typeName: %s, millis: %d%n",
-                            column.name(), column.typeName(), millis);
-                switch (column.typeName().toLowerCase()) {
-                    case "time":
-                        return this.simpleTimeFormatter.format(dateObject);
-                    case "date":
-                        return this.simpleDateFormatter.format(dateObject);
-                    default:
-                        return this.simpleDatetimeFormatter.format(dateObject);
+                            "[TimestampConverter.converterFor] Before converting. rawValue: %s, isTime: %s",
+                            stringValue, isTime);
+
+                String result;
+                try {
+                    result = GetDateTimeFormat(stringValue, column);
+                }
+                catch (Exception e){
+                    System.out.printf(
+                            "[TimestampConverter.converterFor] ERROR! Using regex for conversion. rawValue: %s, " +
+                                    "isTime: %s", stringValue, isTime);
```

Comment:
If it's like C++(printf) isn't isTime Bool?
If so shouldn't it be like so?
```suggestion
                            "[TimestampConverter.converterFor] ERROR! Using regex for conversion. rawValue: %s, " +
                                    "isTime: %b", stringValue, isTime);
```

Reply:
nope :) Java does "to string" to bools; AFAIK %s 
I might be wrong though